### PR TITLE
Fixes a bug in charmd.

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

File: cmd/charmd/main.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func serve(confPath string) error {
 		return errgo.Mask(err)
 	}
 	err = addPublicKey(cache, conf.TermsLocation, conf.TermsPublicKey)
```
Comment (cmars) [inline cmd/charmd/main.go]: I'd rather call `addPublicKey` only if `conf.TermsLocation != ""`.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1370/
Test PASSed.

**Comment by cmars** [comment]
LGTM :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1371/
Test PASSed.

**Pull request merged**